x/nexus/keeper/keeper.go

```diff
@@ -111,9 +134,27 @@ func (k Keeper) SetChain(ctx sdk.Context, chain exported.Chain) {
 }
 
 // LinkAddresses links a sender address to a cross-chain recipient address
-func (k Keeper) LinkAddresses(ctx sdk.Context, sender exported.CrossChainAddress, recipient exported.CrossChainAddress) {
+func (k Keeper) LinkAddresses(ctx sdk.Context, sender exported.CrossChainAddress, recipient exported.CrossChainAddress) error {
+	handler := k.GetRouter().GetRoute(sender.Chain.Module)
+	if handler == nil {
+		return fmt.Errorf("unknown module for sender's chain %s", sender.Chain.String())
+	}
+	if err := handler(ctx, sender); err != nil {
+		return err
+	}
+
+	handler = k.GetRouter().GetRoute(recipient.Chain.Module)
+	if handler == nil {
+		return fmt.Errorf("unknown module for recipient's chain %s", recipient.Chain.String())
+	}
+	if err := handler(ctx, recipient); err != nil {
+		return err
+	}
```

Comment:
just checking with the rest of the team: do we want to force every module to implement a handler (the answer may be yes)